lib/rubocop/cop/rspec/named_subject.rb

```diff
@@ -41,7 +41,44 @@ module RSpec
       #
       #     it { is_expected.to be_valid }
       #   end
+      #
+      # @example `EnforcedStyle: named_only`
+      #   # bad
+      #   RSpec.describe User do
+      #     subject(:user) { described_class.new }
+      #
+      #     it 'is valid' do
+      #       expect(subject.valid?).to be(true)
+      #     end
+      #   end
```

Comment:
Wondering how the cop will behave with a nested context:
```ruby
RSpec.describe User do
  subject(:user) { described_class.new }
  describe 'age' do
    subject { user.age }
    it 'allows driving a car' do
      expect(subject.can_drive_a_car).to be(true)
    end
  end
end
```

Reply:
No offence is detected 👍 

Would you please an example to the spec to indicate that it's as by design?

Reply:
Thanks! Sure, I have added new example `ignores subject where the closest declaration is not named` to the specs